### PR TITLE
feat: LanguageDialog tooltips with translation completeness

### PR DESCRIPTION
**Pull request by buhtz**
The tooltips of each radio button in the LanguageDialog now shows a percent value of how complete the translation is.

- The value is extract from the po-files (using `polib` package) when downloading them from Weblate and integrate them back into our repository.
- The values are stored in a dictionary (indexed by language code) in `languages.py` file.

### CODE REVIEW

File: qt/languagedialog.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 import tools
 import qttools
 import logger
+import languages
```
Comment (aryoda) [inline qt/languagedialog.py:15]: I think this pgk needs to be added to the (optional) dev dependencies in our documentation:

https://github.com/bit-team/backintime/blob/dev/CONTRIBUTING.md#build--install

File: update_language_files.py [hunk 2]
```diff
@@ -2,9 +2,17 @@
 import sys
 import io
 import datetime
+import json
```
Comment (aryoda) [inline update_language_files.py:5]: I think this pgk needs to be added to the (optional) dev dependencies in our documentation:

https://github.com/bit-team/backintime/blob/dev/CONTRIBUTING.md#build--install

File: update_language_files.py [hunk 3]
```diff
@@ -2,9 +2,17 @@
...
 import pprint
```
Comment (aryoda) [inline update_language_files.py:6]: I think this pgk needs to be added to the (optional) dev dependencies in our documentation:

https://github.com/bit-team/backintime/blob/dev/CONTRIBUTING.md#build--install

File: update_language_files.py [hunk 4]
```diff
@@ -2,9 +2,17 @@
...
 from pathlib import Path
 from subprocess import run, check_output
+from common import languages
```
Comment (aryoda) [inline update_language_files.py:9]: I think this pgk needs to be added to the (optional) dev dependencies in our documentation:

https://github.com/bit-team/backintime/blob/dev/CONTRIBUTING.md#build--install

File: update_language_files.py [hunk 5]
```diff
@@ -2,9 +2,17 @@
...
+
+try:
+    import polib
```
Comment (aryoda) [inline update_language_files.py:12]: I think this pgk needs to be added to the (optional) dev dependencies in our documentation:

https://github.com/bit-team/backintime/blob/dev/CONTRIBUTING.md#build--install

File: update_language_files.py [hunk 6]
```diff
@@ -156,13 +168,80 @@ def update_from_weblate():
...
+    """Create dict of language names in different flavours.
+    The dict is used in the LanguageDialog to display the name of
     each language in the UI's current language and the language's own native
     representation.
     """
 
+    # We keep this import local because it is a rare case that this function
+    # will be called. This happens only if a new language is added to BIT.
     try:
         import babel
```
Comment (aryoda) [inline update_language_files.py:246]: I think this pgk needs to be added to the (optional) dev dependencies in our documentation:

https://github.com/bit-team/backintime/blob/dev/CONTRIBUTING.md#build--install

**Review by aryoda — COMMENTED** [review]
Incredibly cool feature, I didn't know how this works!

All my review comments are mainly regarding documentation of packages as (optional) build dependency and perhaps a user dev doc would be great on how to update the translations...

**Comment by buhtz** [comment]
Hi Jürgen,
thanks for reviewing.

> Incredibly cool feature, I didn't know how this works!

Me to, before. 🤣 

> a user dev doc would be great on how to update the translations...

It is IMHO there. Indexed in our [doc-dev TOC](https://github.com/bit-team/backintime/tree/dev/common/doc-dev) titled [Localization and translation using Weblate](https://github.com/bit-team/backintime/blob/dev/common/doc-dev/2_localization.md). But it is my plan to review it again when I do the weblate-sync in the next days. I'm sure there is room for improvment.

**Comment by aryoda** [comment]
> It is IMHO there. Indexed in our [doc-dev TOC](https://github.com/bit-team/backintime/tree/dev/common/doc-dev) titled [Localization and translation using Weblate](https://github.com/bit-team/backintime/blob/dev/common/doc-dev/2_localization.md).

Perfect, I knew there was a documentation but thought it was sphinx-created.

> But it is my plan to review it again when I do the weblate-sync in the next days.

Your translation doc is very clear, just the missing package dependencies and how to update the translation completion in pct could be added (later).

How about linking the doc-dev TOC also in our [contribution.md](https://github.com/bit-team/backintime/blob/dev/CONTRIBUTING.md) to make it more prominently visible? No need to do this in this PR though...

**Review by aryoda — APPROVED** [review]

**Comment by buhtz** [comment]
> How about linking the doc-dev TOC also in our [contribution.md](https://github.com/bit-team/backintime/blob/dev/CONTRIBUTING.md) to make it more prominently visible? No need to do this in this PR though...

The full (2-item) TOC? Mhm... Good idea! Currently the doc-dev link itself is kind of hidden in the [Ressource](https://github.com/bit-team/backintime/blob/dev/CONTRIBUTING.md#resources) section of CONTRIBUTING.md.

**Comment by aryoda** [comment]
> The full (2-item) TOC? Mhm... Good idea! Currently the doc-dev link itself is kind of hidden in the [Ressource](https://github.com/bit-team/backintime/blob/dev/CONTRIBUTING.md#resources) section of CONTRIBUTING.md.

Ah yes :-)

I will open a new issue to centralize the doc-dev to keep this away from this PR

**Pull request merged**